### ISSUE TITLE
Drawing Issues on Raster and Vector Layers

### ISSUE THREAD

**Issue by Jirochian** [opentoonz/opentoonz#3984]
**Question**
Ask a question related to OpenToonz.

Hello, I am not sure what to do about this. When I draw in the canvas on the raster layer on my tablet, my drawing appears far away from where I've drawn it. When I move the camera around, the drawing moves with it and I am not sure if that's supposed to happen. Then when I draw in the vector layer on my tablet, the drawing does not appear until I lift up my pen which makes it kind of difficult to see what I am doing. I found that in the 3D view, there are no problems though, but I would like to figure out if I can use the vector and raster layers still. Both of these problems are nonexistent if I have the window open on my computer and draw with my mouse on the computer, so I figured the problem must be my tablet. However, I tested a painting program on my computer and tablet, and they both performed perfectly and identically, so I've concluded that the problem must be in Opentoonz. I poked around on Github and couldn't quite find a situation like this and I've read several threads about what to do on Reddit for these same problems, but the solutions they found didn't work for me so I was wondering if anyone else could offer some more input. Thank you!

**Comment by RodneyBaker** [comment]
That sounds like an old problem that got fixed.
What release of Opentoonz are you currently using?

**Comment by Jirochian** [comment]
I'm using 1.5; I redownloaded Opentoonz this morning to see if the problem persisted before I sent my question.

**Comment by RodneyBaker** [comment]
Thanks for the information.
Glad to hear you are usng v1.5 but as an aside... I'll do my standard recommendation of downloading and installing the nightly release as that fixes some known problems with lag etc.

For the camera moving part... make sure you have the Camera view on when moving the camera or else you likely will get movement via the canvas/table view.

![image](https://user-images.githubusercontent.com/2576432/122305660-e1178f80-cecc-11eb-840f-27e46470fc46.png)


For other parts of the problem you may need to share a screen recording so that others can deduce what is going on.

**Comment by Jirochian** [comment]
I'll make a recording of the problem right now, but as for downloading the nightly bug fixes, is that the "Download Nightly Built for...?"

**Comment by Jirochian** [comment]
I made about a minute and thirty seconds of recordings but it says the files are too big, do you have any recommendations of how I could send them?

**Comment by manongjohn** [comment]
You may be experiencing an OT issue when you have multiple displays and they are not set up to use the same resolution and scale. Drawing in 1 spot causes the lines to be created in a different area of the canvas is usually a symptom of that.

Can try one of the following:
1. Make sure monitor and tablet use the same resolution (ideal: 1920x1080), or resolution ratio (i.e 16x9). Also check scale is the same on both monitors.
2. Duplicate the main screen to the tablet instead extending it
3. Make the tablet the main screen

You may need to enable Windows Ink support in OT under `Preferences -> Touch / Tablet Settings -> Enable Windows Ink Support.

**Comment by Jirochian** [comment]
Thanks for the tips manongjohn, but I already tried all three of those options to no avail. I wasn't able to get them to match at 1920x1080 because there were only two options for matching resolutions out of all of the potential resolutions, but it still didn't help when I got them to match with the slightly lower res. Also, I am on Mac not Windows, is there a Mac equivalent of your final recommendation? Thanks again!

**Comment by manongjohn** [comment]
No prob. I didnt realize you were on a mac (didnt see it mentioned or missed it) and wasn't sure what you tried.  The last recommendation is for Windows only.

**Comment by Jirochian** [comment]
Okay, thanks again for the help! Sorry if I wasn't totally clear. However, I just messed with settings again to have the tablet and computer mirror screens, but this time I had them both scaled to best fit the computer display as opposed to just matching resolutions or having the best options for the tablet, and it solved the brush lag and problems with vector layer. However new problems surfaced:
- I have to be mirrored all the time and I am not sure if it's supposed to be like that but that seems a little inconvenient; however I'll work with what I got! 
- The tablet screen would have to be shrunk to not fill up the whole screen. Again I can work with this, but I don't think it's supposed to be like that.
- The lines are about half the canvas distance away from where I draw them. This is a big problem and I tried recalibrating the tablet and drawing with the mouse to no avail either.

**Comment by manongjohn** [comment]
What macOS version are you on? What tablet are you using? Might help if others have similar setups or at least help track if there is an issue with certain combinations.

I am not a mac user so any help i can provide is limited. I think there is a way to set scale at an individual application level. Maybe play with those settings?

**Comment by Jirochian** [comment]
I am using macOS Big Sur 11.4, and a Huion Kamvas GT-191. I'll try out the individual scaling. Thanks again!

**Comment by artisteacher** [comment]
As a mac user, I want to suggest checking if the order you connect the tablet and start OT makes a difference. I had similar offset once when I connected to a projector after the program was already running. As long as the projector is connected first and then the app is opened, I don't get offset.

**Comment by Jirochian** [comment]
Thank you for the suggestion artisteacher, but I normally connect the tablet before opening the app and the problem still occurred. I tried it again in case I was mistaken, but to no avail. Thank you though for the tip!

**Comment by ccarterart** [comment]
I'm having a similar issue but on PC with the drawn line not showing up till after I lift my pen up off my tablet (vector layer). Currently using a Wacom Cintiq as my drawing surface/screen and was able to fix the issue with the drawn line showing up a distance from where I drew it on the screen, but also the line isn't showing up at all if I draw on a raster layer. 

At least the vector line shows up even if it's not until I'm done drawing the line, though it does show the line being drawn in real time using 3D view--but that's not ideal for the 2D animating I'm trying to do and would like to be able to use the vector and raster layers.

 I did go ahead and install a clean version of 1.5 (Nightly release version), checked that my graphics drivers were updated (other drawing programs work fine), and still no fix. Should I create a new thread?

**Comment by ccarterart** [comment]
I'm having a similar issue but on PC with the drawn line not showing up till after I lift my pen up off my tablet (vector layer). Currently using a Wacom Cintiq as my drawing surface/screen and was able to fix the issue with the drawn line showing up a distance from where I drew it on the screen, but also the line isn't showing up at all if I draw on a raster layer.

At least the vector line shows up even if it's not until I'm done drawing the line, though it does show the line being drawn in real time using 3D view--but that's not ideal for the 2D animating I'm trying to do and would like to be able to use the vector and raster layers.

I did go ahead and install a clean version of 1.5 (Nightly release version), checked that my graphics drivers were updated (other drawing programs work fine), and still no fix. Should I create a new thread?

**Comment by RodneyBaker** [comment]
The common theme with Huion tablets appears to be bad drivers.
In more than one case a problem has been resolved by REPLACING the current driver.
Note that this does not necessarily mean updating to the latest driver as that has been reported to cause problems in some cases and resolve them in others.  It may be appropriate to dial back to an earlier driver where the current driver causes problems.

Bottom line reported from other Huion users:  Deleting the driver fixes the problem

**Comment by RodneyBaker** [comment]
Closing as likely resolved and/or abandoned after 4 years.
@Please let us know if you are still experiencing this issue.
